.github/workflows/ci.yml

```
@@ -0,0 +1,81 @@
+name: ci
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**.md'
+      - '**.asciidoc'
+  pull_request:
+    paths-ignore:
+      - '**.md'
+      - '**.asciidoc'
+
+# limit the access of the generated GITHUB_TOKEN
+permissions:
+  contents: read
+
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    strategy:
+      matrix:
+        # wait for https://github.com/actions/setup-go/issues/232
+        # to support stable/main
+        go-version: [ "1.15", "1.16", "1.17", "1.18", "1.19" ]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go-version }}
+          check-latest: true
+          cache: true
+      - name: Check
+        run: make check
+
+  coverage:
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+          check-latest: true
+          cache: true
+      - name: Coverage
+        run: |
+          ./scripts/docker-compose-testing run -T --rm trace-context-harness
+          ./scripts/docker-compose-testing up -d --build
+          ./scripts/docker-compose-testing run -T --rm go-agent-tests make coverage GOFLAGS=-v
+
+  test-windows:
+    runs-on: windows-latest
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+          check-latest: true
+          cache: true
+      - name: Build
+        run: go test -v ./...
+
+  test-macos:
+    runs-on: macos-latest
```

Let's try to reproduce the error using a different version

```suggestion
    runs-on: macos-12
```